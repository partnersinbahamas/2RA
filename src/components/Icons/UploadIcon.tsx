import React from 'react';
import PropTypes from 'prop-types';
import Svg from './Svg';

type TProps = {
  className?: string;
  color?: string;
  style?: { [cssAttribute: string]: string };
};

/**
 * @deprecated use SmoothUploadIcon instead
 */
const UploadIcon: React.FC<TProps> = ({ className, color, style }) => (
  <Svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 32 32"
    style={style}
    fill={color}
    className={className}
  >
    <path
      fill="none"
      d="M32,31c0,0.5-0.5,1-1,1H1c-0.5,0-1-0.5-1-1V1c0-0.5,0.5-1,1-1h30c0.5,0,1,0.5,1,1V31z"
    />
    <polygon points="9,25 9,22 12,22 12,25 20,25 20,22 23,22 23,25 23,28 9,28 " />
    <g>
      <rect x="14.5" y="12" width="3" height="12" />
    </g>
    <polygon points="6.1,13.9 8.1,15.9 16,8 23.9,15.9 25.9,13.9 16,4 " />
  </Svg>
);

UploadIcon.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  style: PropTypes.any,
};

UploadIcon.defaultProps = {
  className: '',
  color: 'currentcolor',
  style: {},
};

export default UploadIcon;
