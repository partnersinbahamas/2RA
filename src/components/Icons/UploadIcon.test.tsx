import React from 'react';
import { render } from '@testing-library/react';
import UploadIcon from './UploadIcon';

test('UploadIcon', () => {
  const { container } = render(<UploadIcon />);
  expect(container).toMatchSnapshot();
});
