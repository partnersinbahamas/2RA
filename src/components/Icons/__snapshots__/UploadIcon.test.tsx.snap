// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadIcon 1`] = `
<div>
  <svg
    class="sc-beySPh jroDkm"
    fill="currentcolor"
    viewBox="0 0 32 32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M32,31c0,0.5-0.5,1-1,1H1c-0.5,0-1-0.5-1-1V1c0-0.5,0.5-1,1-1h30c0.5,0,1,0.5,1,1V31z"
      fill="none"
    />
    <polygon
      points="9,25 9,22 12,22 12,25 20,25 20,22 23,22 23,25 23,28 9,28 "
    />
    <g>
      <rect
        height="12"
        width="3"
        x="14.5"
        y="12"
      />
    </g>
    <polygon
      points="6.1,13.9 8.1,15.9 16,8 23.9,15.9 25.9,13.9 16,4 "
    />
  </svg>
</div>
`;
